Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.cpp

```diff
@@ -63,7 +63,7 @@ WebAuthenticatorCoordinatorProxy::~WebAuthenticatorCoordinatorProxy()
 
 const SharedPreferencesForWebProcess& WebAuthenticatorCoordinatorProxy::sharedPreferencesForWebProcess() const
 {
-    return *m_webPageProxy.legacyMainFrameProcess().sharedPreferencesForWebProcess();
```

Comment:
Can we just add a `WebPageProxy::sharedPreferencesForMainFrameProcess()`? I think we want to move away from using `legacyMainFrameProcess()` (as its name suggested), so we probably don't want to introduce more usage.

Reply:
I don't think that makes sense. We don't wanna be using the main frame's shared preferences for all web content processes in a given page. This likely needs to fetch the shared preferences of the relevant web content process.

Reply:
I don't mean changing the existing behavior, but just adding 
```
const SharedPreferencesForWebProcess& WebPageProxy::sharedPreferencesForMainFrameProcess() {
    return legacyMainFrameProcess().sharedPreferencesForWebProcess();
}
```
and changing this to:
```
return  m_webPageProxy.sharedPreferencesForMainFrameProcess();
```
So the message receivers does no need to invoke `legacyMainFrameProcess` on their own.

Reply:
I know what you're suggesting but I don't think that makes sense. We should be using the shared preferences of the sender process, not main frame's process.